utilities/ggd/lib/misc.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package lib
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+// ChangedPackages returns all changed packages.
+func ChangedPackages(changedFiles []string) []string {
+	changedPkgsMap := make(map[string]struct{}, len(changedFiles))
+	for _, f := range changedFiles {
+		dir := filepath.Dir(f)
```

Comment:
Is it ok to take the whole directory path? Was thinking we might need to potentially remove `GOPATH` if it's in the prefix so we can ensure we have the correct import path.

Reply:
the changed files are relative paths already. Will add a comment to indicate.

Reply:
Hmm, so then the packages in the import statements are relative as well? For example, if I do a git diff in statsdex the paths of files which have changed will be relative to the base of the repo (e.g. `services/collector/etc`), but if I were to import one of those packages they would use the full Git path (e.g. `<hostname>/infra/statsdex/services/collector/etc`).

Reply:
> if I were to import one of those packages they would use the full Git path (e.g. <hostname>/infra/statsdex/services/collector/etc

I don't follow. Go imports are always `GOPATH` or `repo/vendor/` relative, what situation are they not going to be?